core/src/main/java/org/testcontainers/utility/PathUtils.java

```diff
@@ -21,21 +18,9 @@
      *
      * @param directory path to the directory to delete.
      */
-    public static void recursiveDeleteDir(final @NonNull Path directory) {
+    public static void recursiveDeleteDir(final @NonNull Path directory, DeleteFileVisitor deleteFileVisitor) {
```

Comment:
this class is public, and this change will break backward compatibility. Please make sure that the old method is still in place